neurom/apps/morph_stats.py

```diff
@@ -73,34 +82,77 @@ def _stat_name(feat_name, stat_mode):
     return '%s_%s' % (stat_mode, feat_name)
 
 
-def extract_stats(neurons, config):
+def extract_stats(neurons, config, as_frame=False):
+    """Extract stats from neurons.
+
+    Arguments:
+        neurons: a neuron, population or neurite tree
+        config (dict): configuration dict. The keys are:
+            - neurite_type: a list of neurite types for which features are extracted
+            - neurite: a dictionary {neurite_feature: mode} where:
+                - neurite_feature is a string from NEURITEFEATURES
+                - mode is an aggregation operation provided as a string such as:
+                  ['min', 'max', 'median', 'mean', 'std', 'raw', 'total']
+        as_frame (bool): if yes, returns a dataframe, else a dict
+
+    Returns:
+        The extracted statistics
+
+    Note:
+        An example config can be found at:
+    """
+    if as_frame:
+        return _extract_stats_as_frame(neurons, config)
+    return _extract_stats_as_dict(neurons, config)
+
+
+extract_stats.__doc__ = extract_stats.__doc__ + EXAMPLE_CONFIG + '\n'
+
+
+def _extract_stats_as_frame(neurons, config):
+    del config['neuron']
```

Comment:
why that? in the config you mentionned above, it does not have a 'neuron' entry, so it crashed here

Reply:
`config['neuron']` contains the config for stats an neuron level. These cannot be splitted by neurite type so they can't be incorporated into the dataframe.
I'm not sure how to handle that.

Reply:
```
if 'neuron' in config:
    del config['neuron']
    L.warning(' neuron based config is not handles with to_frame option')
```

Reply:
alternatively, you can add another neurite_type = 'neuron' in the dataframe